src/Covenant/Internal/Type.hs

```diff
@@ -213,7 +262,7 @@ data BuiltinFlatT
 -- Helpers
 
 newtype ScopeBoundary = ScopeBoundary Int
-  deriving (Show, Eq, Ord, Num) via Int
+  deriving (Show, Eq, Ord, Num, Real, Enum, Integral) via Int
```

Comment:
Why is `Real` needed for this? `Integral` I assume is for adding scope boundaries, but `Real` seems _very_ weird here.

Reply:
https://hackage.haskell.org/package/base-4.21.0.0/docs/GHC-Real.html 

```
class (Real a, Enum a) => Integral a where
```
You'd know better than me why `Real` is a superclass of `Integral`. Does not seem immediately obvious to me that it needs to be?

Reply:
I'm not sure either to be honest, but then again GHC's numerical hierarchy has never made sense.